### PR TITLE
Update changelogs for Docker Desktop edge 2.4.2.0

### PR DESCRIPTION
**Pull request by simonferquel**
### Proposed changes

Update the changelog for Docker Desktop Edge 2.4.2.0

cc @stephen-turner @djs55 (far all grpc-fuse related things)

### CODE REVIEW

**Review by stephen-turner — CHANGES_REQUESTED** [review]
I've made a couple of small comments, but also @usha-mandya should advise on how to write the bug fix bullet points, I think they're not written in our standard format.

File: docker-for-windows/edge-release-notes.md [hunk 1]
```diff
@@ -11,6 +11,29 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- [Docker Compose CLI - 0.1.22](https://github.com/docker/compose-cli/releases/tag/v0.1.22)
+- [Linux kernel 5.4.39](https://hub.docker.com/layers/linuxkit/kernel/5.4.39-f39f83d0d475b274938c86eaa796022bfc7063d2/images/sha256-8614670219aca0bb276d4749e479591b60cd348abc770ac9ecd09ee4c1575405?context=explore)
+- [Kubernetes 1.19.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.19.2)
+- [Go 1.15.2](https://github.com/golang/go/issues?q=milestone:Go1.15.2+label:CherryPickApproved)
+
+* Deprecation
+  - Docker Desktop cannot be installed on Windows 1703 (build 15063) anymore.
+
+### Bug fixes and minor changes
+
```
Comment (djs55) [inline docker-for-windows/edge-release-notes.md:29]: I think these grpcfuse changes are relevant on Windows: (although no-one complained about them so the impact might be less)

```suggestion

- When sharing a file into a container (e.g. `docker run -v C:\Users\...\.gitconfig`) don't watch the parent directory. Fixes [docker/for-mac#4981](https://github.com/docker/for-mac/issues/4981).
- Always flush filesystem caches synchronously on container start. See [docker/for-mac#4943](https://github.com/docker/for-mac/issues/4943).
```

File: docker-for-mac/edge-release-notes.md [hunk 2]
```diff
@@ -11,6 +11,30 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+
+- [Docker Engine 20.10.0 beta1](https://github.com/docker/docker-ce/blob/0fc7084265b3786a5867ec311d3f916af7bf7a23/CHANGELOG.md)
+- [Docker Compose CLI - 0.1.22](https://github.com/docker/compose-cli/releases/tag/v0.1.22)
+- [Linux kernel 5.4.39](https://hub.docker.com/layers/linuxkit/kernel/5.4.39-f39f83d0d475b274938c86eaa796022bfc7063d2/images/sha256-8614670219aca0bb276d4749e479591b60cd348abc770ac9ecd09ee4c1575405?context=explore).
+- [Kubernetes 1.19.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.19.2)
+- [Go 1.15.2](https://github.com/golang/go/issues?q=milestone:Go1.15.2+label:CherryPickApproved)
+
+### Bug fixes and minor changes
+
+- When sharing Linux directories (`/var`, `/bin` etc) with containers, avoid watching paths in the host filesystem.
```
Comment (usha-mandya) [inline docker-for-mac/edge-release-notes.md:27]:
```suggestion
- When sharing Linux directories (`/var`, `/bin`, etc) with containers, Docker Desktop avoids watching paths in the host file system.
```

File: docker-for-mac/edge-release-notes.md [hunk 3]
```diff
@@ -11,6 +11,30 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- When sharing a file into a container (e.g. `docker run -v ~/.gitconfig`) don't watch the parent directory. Fixes [docker/for-mac#4981](https://github.com/docker/for-mac/issues/4981).
```
Comment (usha-mandya) [inline docker-for-mac/edge-release-notes.md:28]:
```suggestion
- When sharing a file into a container (e.g. `docker run -v ~/.gitconfig`), Docker Desktop does not watch the parent directory. Fixes [docker/for-mac#4981](https://github.com/docker/for-mac/issues/4981).
```

File: docker-for-mac/edge-release-notes.md [hunk 4]
```diff
@@ -11,6 +11,30 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- gRPC FUSE: fix `chown` when the file is read-only. Fixes `rabbitmq`, see [docker/for-mac#4964](https://github.com/docker/for-mac/issues/4964).
+- gRPC FUSE: generate `ATTRIB` inotify events as well as `MODIFY`. Fixes [docker/for-mac#4962](https://github.com/docker/for-mac/issues/4962).
+- gRPC FUSE: return `EOPNOTSUPP` from `fallocate` for unsupported modes. This fixes `minio`, see [docker/for-mac#4964](https://github.com/docker/for-mac/issues/4964).
```
Comment (usha-mandya) [inline docker-for-mac/edge-release-notes.md:31]:
```suggestion
- gRPC FUSE: return `EOPNOTSUPP` from `fallocate` for unsupported modes. Fixes `minio`. See [docker/for-mac#4964](https://github.com/docker/for-mac/issues/4964).
```

File: docker-for-mac/edge-release-notes.md [hunk 5]
```diff
@@ -11,6 +11,30 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- Fix NFS mounting, see [docker/for-mac#4958](https://github.com/docker/for-mac/issues/4958).
```
Comment (usha-mandya) [inline docker-for-mac/edge-release-notes.md:32]:
```suggestion
- Fixed an issue related to NFS mounting. See [docker/for-mac#4958](https://github.com/docker/for-mac/issues/4958).
```

File: docker-for-mac/edge-release-notes.md [hunk 6]
```diff
@@ -11,6 +11,30 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- Always flush filesystem caches synchronously on container start. See [docker/for-mac#4943](https://github.com/docker/for-mac/issues/4943).
```
Comment (usha-mandya) [inline docker-for-mac/edge-release-notes.md:33]:
```suggestion
- Always flush file system caches synchronously on container start. See [docker/for-mac#4943](https://github.com/docker/for-mac/issues/4943).
```

File: docker-for-mac/edge-release-notes.md [hunk 7]
```diff
@@ -11,6 +11,30 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- Allow symlinks to point outside of shared volumes. Fixes [docker/for-mac#4862](https://github.com/docker/for-mac/issues/4862).
+- Diagnostics: avoid hanging when Kubernetes is in a broken state.
```
Comment (usha-mandya) [inline docker-for-mac/edge-release-notes.md:35]: Do we have a related ticket for this issue?

File: docker-for-windows/edge-release-notes.md [hunk 8]
```diff
@@ -11,6 +11,29 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- Fix an issue that intermittently make the WSL 2 backend fail to start.
+- Fix NFS mounting, see [docker/for-mac#4958](https://github.com/docker/for-mac/issues/4958).
+- Fix an issue with http proxy exclude lists containing entries shuch as localhost or 127.0.0.1, fixes [docker/for-win#8750](https://github.com/docker/for-win/issues/8750).
```
Comment (usha-mandya) [inline docker-for-windows/edge-release-notes.md:32]:
```suggestion
- Fixed an issue with http proxy exclude lists containing entries such as `localhost` or `127.0.0.1`. Fixes [docker/for-win#8750](https://github.com/docker/for-win/issues/8750).
```

File: docker-for-windows/edge-release-notes.md [hunk 9]
```diff
@@ -11,6 +11,29 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- [Linux kernel 5.4.39](https://hub.docker.com/layers/linuxkit/kernel/5.4.39-f39f83d0d475b274938c86eaa796022bfc7063d2/images/sha256-8614670219aca0bb276d4749e479591b60cd348abc770ac9ecd09ee4c1575405?context=explore)
+- [Kubernetes 1.19.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.19.2)
+- [Go 1.15.2](https://github.com/golang/go/issues?q=milestone:Go1.15.2+label:CherryPickApproved)
+
+* Deprecation
+  - Docker Desktop cannot be installed on Windows 1703 (build 15063) anymore.
+
+### Bug fixes and minor changes
+
+- Fix an issue that intermittently make the WSL 2 backend fail to start.
```
Comment (usha-mandya) [inline docker-for-windows/edge-release-notes.md:30]:
```suggestion
- Fixed an issue that can intermittently cause the WSL 2 backend to fail to start.
```

File: docker-for-windows/edge-release-notes.md [hunk 10]
```diff
@@ -11,6 +11,29 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- Fix NFS mounting, see [docker/for-mac#4958](https://github.com/docker/for-mac/issues/4958).
```
Comment (usha-mandya) [inline docker-for-windows/edge-release-notes.md:31]:
```suggestion
- Fixed an issue related to NFS mounting. See [docker/for-mac#4958](https://github.com/docker/for-mac/issues/4958).
```

File: docker-for-windows/edge-release-notes.md (hunk 8 above)
Comment (usha-mandya) [inline docker-for-windows/edge-release-notes.md:32]:
```suggestion
- Fixed an issue with http proxy exclude lists containing entries such as `localhost` or `127.0.0.1`. Fixes [docker/for-win#8750](https://github.com/docker/for-win/issues/8750).
```

File: docker-for-windows/edge-release-notes.md [hunk 11]
```diff
@@ -11,6 +11,29 @@ This page contains information about Docker Desktop Edge releases. Edge releases
...
+- Fix an issue with http proxy exclude lists containing entries shuch as localhost or 127.0.0.1, fixes [docker/for-win#8750](https://github.com/docker/for-win/issues/8750).
+- When WSL integration process unexpectedly stops, the user is now notified and can decide to restart it or not, instead of always try to restart it in a loop.
+- Fix container logs lagging under heavy load. Fixes [docker/for-win#8216](https://github.com/docker/for-win/issues/8216).
```
Comment (usha-mandya) [inline docker-for-windows/edge-release-notes.md:34]:
```suggestion
- Fixed an issue related to container logs that lag under heavy load. Fixes [docker/for-win#8216](https://github.com/docker/for-win/issues/8216).
```

**Comment by usha-mandya** [comment]
> I've made a couple of small comments, but also @usha-mandya should advise on how to write the bug fix bullet points, I think they're not written in our standard format.

Existing docs use '-'  without any indentation for bulleted list, so I guess the current version looks good. I've added a screenshot for clarity.

image

**Comment by stephen-turner** [comment]
> > I've made a couple of small comments, but also @usha-mandya should advise on how to write the bug fix bullet points, I think they're not written in our standard format.
> 
> Existing docs use '-' without any indentation for bulleted list, so I guess the current version looks good. I've added a screenshot for clarity.
> 
> image

Sorry, I was unclear. I meant the use of the paste tense, and not making them sound like instructions to the user. But you've done all that.

File: docker-for-mac/edge-release-notes.md (hunk 7 above)
Comment (simonferquel) [reply docker-for-mac/edge-release-notes.md:35]: Not a public one

**Comment by simonferquel** [comment]
Updated the PR with the given feedback. 
Reminding:
 - check if we updated linuxkit images 
 - waiting for engine bump to succeed

**Comment by simonferquel** [comment]
Updated release date in changelog

**Review by usha-mandya — APPROVED** [review]
LGTM. Thanks @simonferquel

**Pull request merged**